### PR TITLE
Coverbrowser: event for cache refreshing

### PR DESCRIPTION
**Pull request by hius07**
Self-explained.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10956)

### CODE REVIEW

File: frontend/apps/filemanager/filemanagerbookinfo.lua [hunk 1]
```diff
@@ -164,12 +164,7 @@ function BookInfo:show(file, book_props)
...
             self.custom_book_cover = nil
             if self.prop_updated then
-                local ui = self.ui -- Reader
-                        or require("apps/filemanager/filemanager").instance -- FM, Hist/Coll over FM
-                        or require("apps/reader/readerui").instance -- Hist/Coll over Reader
-                if ui.coverbrowser then -- refresh cache db
-                    ui.coverbrowser:deleteBookInfo(file)
-                end
+                UIManager:broadcastEvent(Event:new("DeleteBookInfo", file)) -- refresh coverbrowser cache db
                 UIManager:broadcastEvent(Event:new("BookMetadataChanged", self.prop_updated))
```
Comment (poire-z) [inline frontend/apps/filemanager/filemanagerbookinfo.lua:168]: I seem to remember you told me deletebookinfo had to be done before bookmetadatachanged - which is what is happening here :) but couldn't it be done in the same BookMetadataChanged event ?

Also, no better name than DeleteBookInfo for this event ? DeleteBookInfo will be the specific action the coverbrowser plugin will do - but no other thing will act on this event. With a more generic name, like BookMetadataAboutToBeChanged (or something nicer :), coverbrowser would delete book info, but other code/pluging could do something else.

Comment (hius07) [reply frontend/apps/filemanager/filemanagerbookinfo.lua:168]: > couldn't it be done in the same BookMetadataChanged event ?

I do not know how to arrange the required order of event handling by different modules.

Comment (poire-z) [reply frontend/apps/filemanager/filemanagerbookinfo.lua:168]: Oh, right, that was the reason.
No idea either, and having 2 events is probably the right solution.
I just feel it needs a better name BookMetadataPreChanged, BookMetadataAboutToBeChanged ...

Comment (hius07) [reply frontend/apps/filemanager/filemanagerbookinfo.lua:168]: I cannot find other good idea for the name.
The event is specially for CoverBrowser, sent **after** metadata changed.

Comment (poire-z) [reply frontend/apps/filemanager/filemanagerbookinfo.lua:168]: BookMetadataObsolete ? BookMetadataStalled ?
(meaning: the things you may have are now invalid - another event will come soon with what's been updated)

Comment (NiLuJe) [reply frontend/apps/filemanager/filemanagerbookinfo.lua:168]: If it's a cache we're talking about, the usual technical term to trash it is "invalidate"; so `InvalidateMetadataCache` (I think document/book is fairly obviously implied there and can be elided ;)).

Comment (hius07) [reply frontend/apps/filemanager/filemanagerbookinfo.lua:168]: I like InvalidateMetadataCache but the general question is: should the event name show what happened or what should be done?

Comment (poire-z) [reply frontend/apps/filemanager/filemanagerbookinfo.lua:168]: It should express what happened, _when something specific happened_, so handlers can do the different things that they think should be done after than specific thing happened.
In some cases when no specific thing happened and it's just a continuation of some handling, the event can be an action (ie. UpdateToc, UpdateFooter).
I like InvalidateMetadataCache too.

**Review by poire-z — APPROVED** [review]

**Pull request merged**